sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.searchStudents.e2e-spec.ts

```diff
@@ -393,6 +394,44 @@ describe("StudentInstitutionsController(e2e)-searchStudents", () => {
     };
   };
 
+  it("Should throw forbidden error when the institution type is not BC Public.", async () => {
+    // Arrange
+
+    // Student submitting an application to College C.
+    const { collegeCApplication } = await saveStudentApplicationForCollegeC(
+      appDataSource,
+    );
+
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCApplication.location,
+    );
+
+    const searchPayload = {
+      appNumber: collegeCApplication.applicationNumber,
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // College C is not a BC Public institution.
+    const collegeCInstitutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(collegeCInstitutionUserToken, BEARER_AUTH_TYPE)
+      .expect({
```

Comment:
I believe that most of the time we are also validating the HTTP status code, for instance `.expect(HttpStatus.BAD_REQUEST)`. I just search the codebase for ".expect(HttpStatus." and got 91 results in 29 files.